opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/bond_encoder.rs

```diff
@@ -114,11 +114,11 @@ pub(crate) struct DynamicSchema {
 }
 
 impl DynamicSchema {
-    pub(crate) fn new(name: &str, namespace: &str, fields: Vec<FieldDef>) -> Self {
+    pub(crate) fn new(name: &str, namespace: &str, fields: &[FieldDef]) -> Self {
         Self {
             struct_name: name.to_string(),
             qualified_name: format!("{namespace}.{name}"),
-            fields,
+            fields: fields.to_vec(),
```

Comment:
This cloning is unnecessary. We could use lifetimes for `DynamicSchema` struct. It would be contained and not cascade to other types.

```rust
#[derive(Clone)]
pub(crate) struct DynamicSchema<'a> {
    pub struct_name: String,
    pub qualified_name: String,
    pub fields: &'a [FieldDef],
}
```